lib/protocol/protocol.js

```diff
@@ -451,22 +451,33 @@ async function doJwpProxy (driver, req, res) {
 /**
  * Check a driver command respond and see if the protocol and value was passed in
  * @param {Object} driverRes Response returned by `executeCommand` in an inner driver
- * @throws Will throw an error if driverRes.error is provided
  * @returns {Object} isW3C, isMJSONWP and driver value or null if it isn't parsable
  */
 function parseProtocol (driverRes) {
   if (!_.isPlainObject(driverRes) || !_.isString(driverRes.protocol)) {
     return null;
   }
 
+  const isW3C = driverRes.protocol === BaseDriver.DRIVER_PROTOCOL.W3C;
+  const isMJSONWP = driverRes.protocol === BaseDriver.DRIVER_PROTOCOL.MJSONWP;
   if (_.has(driverRes, 'value')) {
-    let isW3C = driverRes.protocol === BaseDriver.DRIVER_PROTOCOL.W3C;
-    let isMJSONWP = driverRes.protocol === BaseDriver.DRIVER_PROTOCOL.MJSONWP;
     return {isW3C, isMJSONWP, value: driverRes.value};
   }
-
-  if (_.isError(driverRes.error)) {
-    throw driverRes.error;
+  if (_.has(driverRes, 'error')) {
+    if (isMJSONWP) {
+      return {isW3C, isMJSONWP, value: {
+        status: driverRes.error.jsonwpCode,
+        value: driverRes.error.message,
+      }};
+    }
```

Comment:
You don't need to translate the error. The error should be passed along on it's own. The functions `getResponseForW3CError` and `getResponseForMJSONWPError` parses the `Error` and then responds with the status/value.